### ISSUE TITLE
Ability to specify accepted languages for public servers

### ISSUE THREAD

**Issue by davidebeatrici** [mumble-voip/mumble#4742]
Now that we solely rely on GeoIP for the server categorization and dropped `registerLocation` (#3328), we should implement a field that defines the accepted languages.

That way a user can easily distinguish between servers that are closer and servers on which the desired language is effectively accepted.

**Comment by AlD** [comment]
IMHO server country doesn't matter if we have latency. I'd drop server country altogether and only show desired languages and latency.

**Comment by davidebeatrici** [comment]
![](https://user-images.githubusercontent.com/5897523/107433464-a4f0ae80-6b29-11eb-9ce9-43c8ff7c7673.png)

Also, I just noticed a bug: the servers are correctly ordered by latency, but the value in the `Ping` column is not updated.

**Comment by SC1040-TS2** [comment]
Personally, I believe Server Country filtering should be retained as a toggleable feature, given that users may only want to compare the given latencies of servers in one region as is usually the case(Useful for particularly large ones without per-country filters such as the North America US Server Country), as well as given that not all servers are set to the region they are actually hosted in.

Besides, from what activity I have seen in the US region, users may prioritize latency at first, but will eventually just keep going back to servers that are preferred above others.

**Comment by davidebeatrici** [comment]
Of course.

**Comment by toby63** [comment]
> IMHO server country doesn't matter if we have latency. I'd drop server country altogether and only show desired languages and latency.

I disagree, there was discussion about why Mumble should not ping all servers always and therefore the correct country origin should still be visible imo.
At least we need options:
- If ping is disabled, then show country origin.
- show country origin and ping (should be default imo, but thats up to you)

Also there are some more things to consider:
Is it really a good idea to let the server owner choose which country/language they want to be recognized with? I think users should have a way to know where a server really comes from. 

The better compromise would be to:
- show the correct country origin
- make it clear that the language preference is chosen by the server owner

**Sidenote:** We might also discuss **more language distinctions** and how they are displayed; for example on many websites there is a distinction between British english and american english.

There are two reasons for that:
- language differences: including wording, accents etc.
- cultural/country/region differences: This might be a bit problematic, but you cannot deny that language is also closely related to this; for example:
   - some people might expect to talk to people from a more specific region
   - some people might be offended to see only one category or even just one flag for a language (if we would use flags for that)

**Comment by toby63** [comment]
@Krzmbrzl Could you give a bit more detail to your thumps down? I mention several different things.

**Comment by AlD** [comment]
> > IMHO server country doesn't matter if we have latency. I'd drop server country altogether and only show desired languages and latency.
> 
> I disagree, there was discussion about why Mumble should not ping all servers always and therefore the correct country origin should still be visible imo.

It's not like pinging everything constantly is the only alternative. The client could ping all servers once, and then keep updating the top N (e.g. 50/100/configurable).

> Is it really a good idea to let the server owner choose which country/language they want to be recognized with? I think users should have a way to know where a server really comes from.

But they do, just by looking at the ping. I don't think any user in e.g. Europe really cares if a server is hosted in a neighboring country. Just which languages are preferred there and whether it's within a reasonable network latency matters.

**Comment by toby63** [comment]
I think a lot people will not like it, when you suddenly remove the country categories.
I can only strongly advice against such a step.

**Comment by AlD** [comment]
Ok, but what's the use case you have in mind? So far you brought up technical concerns (pinging everything is bad), and maybe a sort of trust concern ("I think users should have a way to know where a server really comes from"), not sure tbh.

**Comment by toby63** [comment]
You named a good example:
Canada.
You could have a fantastic combination of country (canada) and language (french or english) to identify a server for you.
Also an interesting option would be, to be able to choose more than just one language.

In general the country category is established (also outside of mumble).

**Comment by Krzmbrzl** [comment]
> Could you give a bit more detail to your thumps down? I mention several different things.

I will not have the pinging discussion again. This was held elsewhere already and we simply have completely different opinions on that topic.
Therefore I do not follow your argument of not pinging servers, making the whole argument afterwards superfluous.

And I also don't like the ideas presented afterwards either. That's why I simply downvoted.

> I think a lot people will not like it, when you suddenly remove the country categories.
I can only strongly advice against such a step.

That has already been done and was decided upon. But rejection of yours is noted.
The country information is still available though, so if you care you still have it. You just don't have to click yourself through the mess of a country tree in order to see which servers are available.

A further note: The country categories are not topic of this issue. So please don't continue arguing about that here. If you feel like this requires some more discussion, please go ahead and create a new discussion for it.

**Comment by toby63** [comment]
@Krzmbrzl Ok, I was not aware that countries were already removed.
I think it is closely related to this, but everything is already said, so I will wait for further comments.
